src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -680,7 +682,7 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
         let result = match ns {
             Namespace::MacroNS => self.resolve_macro(path_str, module_id).map_err(ErrorKind::from),
             Namespace::TypeNS | Namespace::ValueNS => {
-                self.resolve(path_str, ns, module_id, extra_fragment).map(|(res, _)| res)
+                self.resolve(path_str, ns, module_id, extra_fragment, None).map(|(res, _)| res)
```

Comment:
I didn't passed `self_id` to `check_full_res`, because it had no role in solving the bug, so it call `resolve` with `None` instead. But maybe we should ?

Reply:
I think we should; can you add `self_id` as a parameter to `check_full_res`? You may have to also add it to some of the error variants.

The worst that happens is the diagnostic is wrong though, so if it's complicated it's ok to save it for a follow-up. I think this is a case that would be affected:

```rust
struct Foo {
	foo: i32,
}

mod bar {
	impl crate::Foo {
		/// Baz the [macro@Self::foo].
		pub fn baz(&self) {}
	}
}
```